test/runtests.jl

```diff
@@ -197,6 +199,80 @@ end
     @test @inferred(f9(plan_fft(zeros(10), 1), 10)) == 1/10
 end
 
+@testset "output size" begin
+    @testset "complex fft output size" begin
+        for x in (randn(3), randn(3, 4), randn(3, 4, 5))
+            N = ndims(x)
+            y = randn(size(x))
+            for dims in unique((1, 1:N, N))
+                P = plan_fft(x, dims)
+                @test AbstractFFTs.output_size(P) == size(x)
+                @test AbstractFFTs.output_size(P') == size(x)
+                Pinv = plan_ifft(x)
+                @test AbstractFFTs.output_size(Pinv) == size(x)
+                @test AbstractFFTs.output_size(Pinv') == size(x)
+            end
+        end
+    end
+    @testset "real fft output size" begin
+        for x in (randn(3), randn(4), randn(3, 4), randn(3, 4, 5)) # test odd and even lengths
+            N = ndims(x)
+            for dims in unique((1, 1:N, N))
+                P = plan_rfft(x, dims)        
+                Px_sz = size(P * x)
+                @test AbstractFFTs.output_size(P) == Px_sz 
+                @test AbstractFFTs.output_size(P') == size(x) 
+                y = randn(Px_sz) .+ randn(Px_sz) * im
+                Pinv = plan_irfft(y, size(x)[first(dims)], dims)
+                @test AbstractFFTs.output_size(Pinv) == size(Pinv * y)
+                @test AbstractFFTs.output_size(Pinv') == size(y)
+            end
+        end
+    end
+end
+
+@testset "adjoint" begin
+    @testset "complex fft adjoint" begin
+        for x in (randn(3), randn(3, 4), randn(3, 4, 5))
+            N = ndims(x)
+            y = randn(size(x))
+            for dims in unique((1, 1:N, N))
+                P = plan_fft(x, dims)
+                @test (P')' * x == P * x # test adjoint of adjoint
+                @test size(P') == AbstractFFTs.output_size(P) # test size of adjoint 
+                @test dot(y, P * x) ≈ dot(P' * y, x) # test validity of adjoint
+                @test dot(y, P \ x) ≈ dot(P' \ y, x)
+                Pinv = plan_ifft(y)
+                @test (Pinv')' * y == Pinv * y 
+                @test size(Pinv') == AbstractFFTs.output_size(Pinv) 
+                @test dot(x, Pinv * y) ≈ dot(Pinv' * x, y)
+                @test dot(x, Pinv \ y) ≈ dot(Pinv' \ x, y)
+                @test_throws MethodError mul!(x, P', y)
+            end
+        end
+    end
+    @testset "real fft adjoint" begin
+        for x in (randn(3), randn(4), randn(3, 4), randn(3, 4, 5)) # test odd and even lengths
+            N = ndims(x)
+            for dims in unique((1, 1:N, N))
+                P = plan_rfft(x, dims)        
+                y_real = randn(size(P * x))
+                y_imag = randn(size(P * x))
+                y = y_real .+ y_imag .* im 
```

Comment:
You can also just use
```suggestion
                y = randn(Complex{Float64}, size(P * x))
```